controllers/capabilities/openapi_controller.go

```diff
@@ -261,13 +300,108 @@ func (r *OpenAPIReconciler) checkProductSynced(resource *capabilitiesv1beta1.Ope
+	openapiRefFldPath := specFldPath.Child("openapiRef")
+	secretRefFldPath := openapiRefFldPath.Child("secretRef")
+
+	oasSourceSecret := &corev1.Secret{}
+	objectKey := types.NamespacedName{Name: openAPICR.Spec.OpenAPIRef.SecretRef.Name, Namespace: openAPICR.Spec.OpenAPIRef.SecretRef.Namespace}
+
+	// Read the OAS source secret
+	if err := r.Client().Get(r.Context(), objectKey, oasSourceSecret); err != nil {
+		if errors.IsNotFound(err) {
+			fieldErrors = append(fieldErrors, field.Invalid(secretRefFldPath, openAPICR.Spec.OpenAPIRef.SecretRef, "Secret not found"))
+			return &helper.SpecFieldError{
+				ErrorType:      helper.InvalidError,
+				FieldErrorList: fieldErrors,
+			}
+		}
+		// Unexpected error
+		return err
+	}
+
+	// Add label to OAS source secret so it can be watched
+	if oasSourceSecret.ObjectMeta.Labels == nil {
+		oasSourceSecret.ObjectMeta.Labels = map[string]string{}
+	}
+	oasSourceSecret.ObjectMeta.Labels[oasSecretLabelSelectorKey] = oasSecretLabelSelectorValue
+	if err := r.Client().Update(r.Context(), oasSourceSecret); err != nil {
+		return err
+	}
+
+	// Re-fetch the OpenAPI CR in case it's been modified
+	objectKey = types.NamespacedName{Name: openAPICR.Name, Namespace: openAPICR.Namespace}
+	if err := r.Client().Get(r.Context(), objectKey, openAPICR); err != nil {
+		return err
+	}
+
+	// Add label to OpenAPI CR with source secret's UID
+	if openAPICR.ObjectMeta.Labels == nil {
+		openAPICR.ObjectMeta.Labels = map[string]string{}
+	}
+	openAPICR.ObjectMeta.Labels[openAPISecretRefLabelKey] = string(oasSourceSecret.GetUID())
+	if err := r.Client().Update(r.Context(), openAPICR); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *OpenAPIReconciler) annotateOpenAPICR(openAPICR *capabilitiesv1beta1.OpenAPI) error {
+	// Re-fetch the OpenAPI CR in case it's been modified
+	objectKey := types.NamespacedName{Name: openAPICR.Name, Namespace: openAPICR.Namespace}
+	if err := r.Client().Get(r.Context(), objectKey, openAPICR); err != nil {
+		return err
+	}
+
+	if openAPICR.Annotations == nil {
+		openAPICR.Annotations = map[string]string{}
+	}
+
+	// Annotate the CR with a default requeue interval; this is so we can check the source OAS URL for changes
+	// A default interval is provided but we allow the user to override it
+	intervalStr, ok := openAPICR.Annotations[pollOasURLIntervalAnnotation]
+	if !ok {
+		openAPICR.Annotations[pollOasURLIntervalAnnotation] = defaultPollOasURLInterval
+		if err := r.Client().Update(r.Context(), openAPICR); err != nil {
+			return err
+		}
+	} else {
+		// If the user provided interval is invalid, reset it to the default value
+		// This is to ensure a valid interval exists and to let the user know that their interval couldn't be parsed
+		_, err := time.ParseDuration(intervalStr)
+		if err != nil {
+			openAPICR.Annotations[pollOasURLIntervalAnnotation] = defaultPollOasURLInterval
+			if err2 := r.Client().Update(r.Context(), openAPICR); err2 != nil {
```

Comment:
isn't that if we are not doing anything with err but using it just to update the annotation we don't need it and therefore the above can be again - err? 
It looks like we could really use admission webhook here I think, it might be too much though...
Just wonder if there's anything we could do here that would communicate to users about the error. 
Also I'm not sure I like the idea of operator reverting it to default when wrong :/

Reply:
I removed this function altogether because we decided to remove functionality to let the user specify the OAS URL scrape interval.